paragon/core/services/sprites.py

```diff
@@ -16,8 +18,61 @@ def __init__(self, gd):
         }
         self.team_names = ["青", "赤", "緑", "紫"]
 
-    def from_spawn(self, spawn, person_key=None) -> Optional[QPixmap]:
-        team = 0
+        self.timer = QTimer()
+        self.sprite_items = list()
+        self.activated = list()
+        self.timer.timeout.connect(self._next_frame)
+
+    def add_sprite_to_handler(self, sprite_item: SpriteItem):
+        if self.timer.isActive():
+            self.activated.append(QDateTime().currentMSecsSinceEpoch())
+        self.sprite_items.append(sprite_item)
+
+    def delete_sprite_from_handler(self, sprite_item: SpriteItem):
+        if self.activated and self.sprite_items:
+            for index in range(len(self.sprite_items)):
+                if self.sprite_items[index] == sprite_item:
+                    self.sprite_items.pop(index)
+                    self.activated.pop(index)
+                    break
+
+    def start_handler(self):
+        time = QDateTime().currentMSecsSinceEpoch()
+        self.activated = [time for _ in range(len(self.sprite_items))]
+        # Check every 30Hz or 30FPS for layman's terms
+        self.timer.start(1000/30)
+    
+    def stop_handler(self):
+        self.timer.stop()
+
+    def _next_frame(self):
+        current_time = QDateTime().currentMSecsSinceEpoch()
+        for x in range(len(self.sprite_items)):
+            # If sprite is loaded
+            if sprite := self.sprite_items[x].sprite:
```

Comment:
This chain of if/else feels like something the sprite items should be responsible for.